### PR TITLE
feat(provider/kubernetes): V2 operation init

### PR DESCRIPTION
**Pull request by lwander**
Continuation of https://github.com/spinnaker/clouddriver/pull/1820

spinnaker/spinnaker#1902
spinnaker/spinnaker#1916
spinnaker/spinnaker#1922

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/KubernetesApiVersion.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.kubernetes.v2.description;
+
+import java.util.Arrays;
+
+public enum KubernetesApiVersion {
+  V1("v1"),
+  EXTENSIONS_V1BETA1("extensions/v1beta1");
```
Comment (jtk54) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/KubernetesApiVersion.java]: Does this correspond to v2 of the Kubernetes provider? Will there be a mapping defined between Kubernetes API version and provider version?

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/converter/KubernetesDeployManifestConverter.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation;
+import com.netflix.spinnaker.clouddriver.security.AbstractAtomicOperationsCredentialsSupport;
+import com.netflix.spinnaker.clouddriver.security.ProviderVersion;
+
+import java.util.Map;
+
+public class KubernetesDeployManifestConverter extends AbstractAtomicOperationsCredentialsSupport {
+  @Override
+  public AtomicOperation convertOperation(Map input) {
+    throw new RuntimeException("Not implemented");
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/converter/KubernetesDeployManifestConverter.java]: Could throw this: https://docs.oracle.com/javase/7/docs/api/java/lang/UnsupportedOperationException.html

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/KubernetesApiVersion.java (hunk 1 above)
Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/KubernetesApiVersion.java]: No these are Kubernetes-only release tracks. The client library is broken down by operations on each of these, so we need to be able to pull the right release track from a manifest to know how to update/deploy/destroy it.

We want to support all of these, so no mapping is really needed. What is important will be to connect Spinnaker versions to Kubernetes versions (both separate from provider version and kubernetes api version entirely).

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/converter/KubernetesDeployManifestConverter.java (hunk 2 above)
Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/converter/KubernetesDeployManifestConverter.java]: SGTM

**Pull request merged**